qualtran/surface_code/quantum_error_correction_scheme.py

```diff
@@ -0,0 +1,90 @@
+#  Copyright 2023 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import attr
+import numpy as np
+from attrs import frozen
+
+from qualtran.surface_code.physical_parameters import PhysicalParameters
+
+
+@frozen
+class QuantumErrorCorrectionScheme:
+    """QuantumErrorCorrectionScheme represents a quantum error correction scheme.
```

Comment:
this is an unhelpful line.

A full quantum error correction scheme would include a particular code, decoder, way of doing gates, and many more things. As I understand it: this class captures a small slice of the relevant characteristics of an error correcting code. Consider a more helpful name for this class. And/or consider making this abstract and giving it a concise name (and meaningful names to the implementations).